src/RecipeRepository.js

```diff
@@ -0,0 +1,57 @@
+class RecipeRepository {
+  constructor(recipeData) {
+    this.recipes = recipeData
+  }
+
+  findRecipeTags () {
+    let tags = [];
+    this.recipes.forEach(recipe => {
+      recipe.tags.forEach(tag => {
+        if (!tags.includes(tag)) {
+          tags.push(tag);
+        }
+      });
+    });
+    tags.sort();
+    return tags;
+  }
+
+  findTaggedRecipes (selectedTags) {
+    let filteredTagResults = [];
+    selectedTags.forEach(tag => {
+      let allRecipes = this.recipes.filter(recipe => {
+        return recipe.tags.includes(tag);
+      });
+
+      allRecipes.forEach(recipe => {
+        if (!filteredTagResults.includes(recipe)) {
+          filteredTagResults.push(recipe);
+        }
+      })
+    });
+   return filteredTagResults;
+  }
+
+  //might need to add functionality to lowercase all names/search material
```

Comment:
agreed ..there is the function in the scripts file to do this so we can see how to make these talk to each other